ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/ColorStyle.kt

```diff
@@ -0,0 +1,201 @@
+import dev.drewhamilton.poko.Poko
+import kotlin.math.abs
+import kotlin.math.cos
+import kotlin.math.max
+import kotlin.math.sin
+
+internal sealed interface ColorStyle {
+
+    @JvmInline
+    value class Solid(@JvmSynthetic val color: Color) : ColorStyle
+
+    @JvmInline
+    value class Gradient(@JvmSynthetic val brush: Brush) : ColorStyle
+}
+
+@JvmSynthetic
+@Composable
+internal fun ColorScheme.toColorStyle(): ColorStyle {
+    val colorInfo = if (isSystemInDarkTheme()) dark ?: light else light
+
+    return when (colorInfo) {
+        is ColorInfo.Alias -> TODO("Color aliases are not yet implemented.")
+        is ColorInfo.Hex -> ColorStyle.Solid(Color(color = colorInfo.value))
+        is ColorInfo.Gradient -> ColorStyle.Gradient(
+            when (colorInfo) {
+                is ColorInfo.Gradient.Linear -> relativeLinearGradient(
+                    colorStops = colorInfo.points.toColorStops(),
+                    degrees = colorInfo.degrees,
+                )
+
+                is ColorInfo.Gradient.Radial -> Brush.radialGradient(
+                    colorStops = colorInfo.points.toColorStops(),
+                )
+            },
+        )
+    }
+}
+
+private fun List<ColorInfo.Gradient.Point>.toColorStops(): Array<Pair<Float, Color>> =
+    map { point -> point.percent to Color(point.color) }
+        .toTypedArray()
+
+@Stable
+private fun relativeLinearGradient(
+    vararg colorStops: Pair<Float, Color>,
+    degrees: Float,
+    tileMode: TileMode = TileMode.Clamp,
+): ShaderBrush {
+    val end = RelativeEndOffset(degrees)
+    return relativeLinearGradient(
+        colorStops = colorStops,
+        start = Offset(x = 1f, y = 1f) - end,
+        end = end,
+        tileMode = tileMode,
+    )
+}
+
+@Stable
+private fun relativeLinearGradient(
+    vararg colorStops: Pair<Float, Color>,
+    start: Offset,
+    end: Offset,
+    tileMode: TileMode = TileMode.Clamp,
+): ShaderBrush =
+    RelativeLinearGradient(
+        colors = List(colorStops.size) { i -> colorStops[i].second },
+        stops = List(colorStops.size) { i -> colorStops[i].first },
+        start = start,
+        end = end,
+        tileMode = tileMode,
+    )
+
+@Suppress("FunctionName")
+private fun RelativeEndOffset(degrees: Float): Offset {
+    // Convert the angle to radians.
+    val radians = Math.toRadians(degrees.toDouble())
+
+    // Calculate the normalized x and y positions.
+    val xPosition = cos(radians)
+    val yPosition = sin(radians)
+
+    // Determine the scaling factor to move the point to the edge of the enclosing square.
+    val scaleFactor = max(abs(xPosition), abs(yPosition))
+
+    // Scale the x and y coordinates.
+    val scaledX = xPosition / scaleFactor
+    val scaledY = yPosition / scaleFactor
+
+    // Convert the scaled coordinates to an Offset.
+    val x = (scaledX + 1) / 2f
+    val y = (1 - scaledY) / 2f
+
+    return Offset(x.toFloat(), y.toFloat())
```

Comment:
Thanks to @MarkVillacampa for the inspiration 🙏 

Note that in this implementation the angle is only correct when the drawing area is a square. We're still discussing our expected result. If that ends up being different, I'll probably do that in a follow-up PR.